src/plugins/intel_cpu/src/nodes/fullyconnected.cpp

```diff
@@ -39,19 +42,22 @@ namespace node {
 bool FullyConnected::isSupportedOperation(const std::shared_ptr<const ov::Node>& op,
                                           std::string& errorMessage) noexcept {
     try {
-        const auto fc = std::dynamic_pointer_cast<const FullyConnectedNode>(op);
-        if (!fc) {
-            errorMessage = "Only legacy FullyConnected operation is supported";
+        if (!ov::is_type<const ov::op::internal::FullyConnected>(op) &&
+            !ov::is_type<const ov::op::internal::FullyConnectedQuantizedLegacy>(op) &&
+            !ov::is_type<const ov::op::internal::FullyConnectedCompressed>(op)) {
             return false;
         }
-        if (fc->get_input_size() == 3 &&
-            std::dynamic_pointer_cast<const ov::op::v0::Constant>(fc->get_input_node_shared_ptr(BIAS_ID)) == nullptr) {
-            errorMessage = "Only Constant operation on 'bias' input is supported";
+
+        const auto fc = std::dynamic_pointer_cast<const ov::op::internal::FullyConnected>(op);
+        if (!fc) {
+            errorMessage = "Only FullyConnected operation is supported";
             return false;
         }
-        const auto weightRank = fc->get_input_partial_shape(WEIGHTS_ID).size();
-        if (weightRank != 2) {
-            errorMessage = "Doesn't support 'weight' input with rank: " + std::to_string(weightRank);
+
+        if (fc->get_input_size() == 3 &&
+            (std::dynamic_pointer_cast<const ov::op::v0::Constant>(fc->get_input_node_shared_ptr(BIAS)) == nullptr &&
+             std::dynamic_pointer_cast<const ov::op::internal::Placeholder>(fc->get_input_node_shared_ptr(BIAS)) == nullptr)) {
```

Comment:
We can write much shorter:
```suggestion
            (!ov::is_type<const ov::op::v0::Constant>(fc->get_input_node_shared_ptr(BIAS)) &&
             !ov::is_type<const ov::op::internal::Placeholder>(fc->get_input_node_shared_ptr(BIAS)))) {
```